src/components/AddItemModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import DefaultModal, { Overlay, ButtonField } from './DefaultModal';
+import Logo from '../molecules/Logo';
+import TextInput from './TextInput';
+import CancelButton from './CancelButton';
+import MainButton from './MainActionButton';
+
+const Body = styled.div`
+	height: 20em;
+`;
+
+const Desc = styled.p`
+	color: black;
+	font-size: 14px;
+`;
+
+const BodyLogo = styled.div`
+	grid-area: logo;
+	justify-content: center;
+	align-items: center;
+`;
+
+const InputField = styled.div`
+	grid-area: input;
+	margin: 0 2em;
+	display: grid;
+	grid-template-columns: 1fr 1fr;
+	grid-template-rows: 1fr 1.5fr;
+	grid-template-areas:
+		'name date'
+		'weight grade'
+	color: black;
+	justify-content: center;
+	align-items: center;
+`;
+
+const InputName = styled(TextInput)`
+	grid-area: name;
+`;
+
+const InputDate = styled(TextInput)`
+	grid-area: date;
+`;
+
+const InputWeight = styled(TextInput)`
+	grid-area: weight;
+`;
+
+const InputGrade = styled(TextInput)`
+	grid-area: grade;
+`;
+
+const Cancel = styled(CancelButton)`
+	grid-area: left;
+`;
+
+const Submit = styled(MainButton)`
+	grid-area: right;
+`;
+
+export interface AddItemInterface {
+	onSubmit: (data: AddItemData) => void;
+}
+
+export interface AddItemData {
+	name: string;
+	date: string;
+	weight: string;
+	grade: string;
+}
+
+function AddItemModal({ onSubmit }: AddItemInterface) {
+	const [open, setOpen] = useState(true);
+	const [name, setName] = useState('');
+	const [date, setDate] = useState('');
+	const [weight, setWeight] = useState('');
+	const [grade, setGrade] = useState('');
+
+	function close() {
+		setOpen(false);
+	}
+
+	function handleSubmit({ name, date, weight, grade }: AddItemData) {
+		onSubmit({ name, date, weight, grade });
+		close();
+	}
+
+	return (
+		<Overlay onClick={close} open={open}>
```

Comment:
Overlay should live in the default modal. It should be the root element of the default modal. I think the user of default modal shouldn't have to think about the overlay but merely the actual contents of the modal (header, footer, body).

Changes here that are kind of related: Let's assume the following components

Top component: Call it `<Page />`
Add Item Modal
Default Modal

In the page component, it will have the state variable `addItemOpen` which it use to decide if it will render the AddItemModal (using `{ addItemOpen && <AddItemModal />}` style syntax). As far as the `AddItemModal` is concerned, it is always visible if rendered (no display none, no conditional rendering inside it or the default modal at all and it won't need an `isOpen` prop.)

AddItemModal will have a onSubmit (we can maybe change the prop to handleSubmit and the handleSubmit inside the component to onSubmit) function and a handleClose function as props. It will call the close function wherever we expect to have closing behaviour. It is assumed that the passed handleClose function will modify the state in <Page /> which will translate to the modal being hidden. 

The default modal will also need to use the handleClose function to handle clicking on the overlay. It probably won't need a handle submit since you create those buttons in the AddItemModal and the relevant state. 

The reason we need to do it like this, is because the parent needs to be able to control the modal (opening and closing it) because it will be responsible for mapping a button outside the scope of this component to opening the modal.

If anything I said here doesn't make sense, or you don't understand why I think we should make this change / might have to, let me know and I am happy to clarify and walk through the specifics.

Reply:
So after these changes, `useState` in terms of open/closed will be entirely up to the user of the modal?

Reply:
And thanks a ton! I believe this makes sense, though how would the user go about changing the visibility of the modal without digging into the file? Would they extend the CSS in `DefaultModal.tsx` and use classnames from there?